### PR TITLE
UUID for wallets

### PR DESCRIPTION
**Pull request by marcosrdz**
Need this for upcoming features. It will allow me to reference a wallet via random ID instead of encrypting the value of a wallet's secret. Of course. MD5 is out of the question.

### CODE REVIEW

**Comment by marcosrdz** [comment]
Updated.

File: class/abstract-wallet.js [hunk 1]
```diff
@@ -29,6 +30,7 @@ export class AbstractWallet {
     this.preferredBalanceUnit = BitcoinUnit.BTC;
     this.chain = Chain.ONCHAIN;
     this.hideBalance = false;
+    this.id = undefined;
```
Comment (Overtorment) [inline class/abstract-wallet.js]: you dont need to save wallet's id if you easily generate it on the fly. remove

File: class/abstract-wallet.js [hunk 2]
```diff
@@ -112,5 +114,18 @@ export class AbstractWallet {
     return 0;
   }
 
+  setID() {
```
Comment (Overtorment) [inline class/abstract-wallet.js]: rename it to `getId` and remove `if (this.id === undefined) {`. and also pls change `rmd160` to `sha256`, so its double sha256 hash (more secure)

File: class/app-storage.js [hunk 3]
```diff
@@ -96,7 +96,7 @@ export class AppStorage {
       return false;
     }
 
-    return !!data;
+    return data === true || data === '1';
```
Comment (Overtorment) [inline class/app-storage.js]: remove

File: class/app-storage.js [hunk 4]
```diff
@@ -184,6 +184,7 @@ export class AppStorage {
         data = JSON.parse(data);
         if (!data.wallets) return false;
         let wallets = data.wallets;
+        let isWalletWithoutID = false;
```
Comment (Overtorment) [inline class/app-storage.js]: remove

File: class/app-storage.js [hunk 5]
```diff
@@ -239,13 +240,22 @@ export class AppStorage {
               break;
           }
           // done
-          if (!this.wallets.some(wallet => wallet.getSecret() === unserializedWallet.secret)) {
+          if (unserializedWallet.id === undefined) {
```
Comment (Overtorment) [inline class/app-storage.js]: remove

File: class/app-storage.js [hunk 6]
```diff
@@ -239,13 +240,22 @@ export class AppStorage {
...
+            isWalletWithoutID = true;
+            unserializedWallet.setID();
+          }
+          if (!this.wallets.some(wallet => wallet.id === unserializedWallet.id)) {
             this.wallets.push(unserializedWallet);
             this.tx_metadata = data.tx_metadata;
           }
         }
+        if (isWalletWithoutID) {
```
Comment (Overtorment) [inline class/app-storage.js]: remove

File: screen/wallets/add.js [hunk 7]
```diff
@@ -256,6 +256,7 @@ export default class WalletsAdd extends Component {
                             }
                             A(A.ENUM.CREATED_LIGHTNING_WALLET);
                             await w.generate();
+                            w.setID();
```
Comment (Overtorment) [inline screen/wallets/add.js]: remove

File: screen/wallets/add.js [hunk 8]
```diff
@@ -306,6 +307,7 @@ export default class WalletsAdd extends Component {
                         }
                         if (this.state.activeBitcoin) {
                           await w.generate();
+                          w.setID();
```
Comment (Overtorment) [inline screen/wallets/add.js]: remove

File: screen/wallets/import.js [hunk 9]
```diff
@@ -58,11 +58,13 @@ export default class WalletsImport extends Component {
 
   async _saveWallet(w) {
     if (BlueApp.getWallets().some(wallet => wallet.getSecret() === w.secret)) {
+      ReactNativeHapticFeedback.trigger('notificationError', { ignoreAndroidSystemSettings: false });
       alert('This wallet has been previously imported.');
     } else {
       alert(loc.wallets.import.success);
       ReactNativeHapticFeedback.trigger('notificationSuccess', { ignoreAndroidSystemSettings: false });
       w.setLabel(loc.wallets.import.imported + ' ' + w.typeReadable);
+      w.setID();
```
Comment (Overtorment) [inline screen/wallets/import.js]: remove

File: screen/wallets/walletMigrate.js [hunk 10]
```diff
@@ -21,7 +21,7 @@ export default class WalletMigrate extends Component {
     try {
       const data = await AsyncStorage.getItem('data');
       if (data) {
-        const isEncrypted = (await AsyncStorage.getItem('data_encrypted')) || '';
+        const isEncrypted = (await AsyncStorage.getItem('data_encrypted')) || '0';
```
Comment (Overtorment) [inline screen/wallets/walletMigrate.js]: remove

**Pull request closed without merging**